system/jlib/jmutex.hpp

```diff
@@ -288,7 +288,15 @@ class CriticalSection
     inline ~CriticalSection()
     {
 #ifdef _ASSERT_LOCK_SUPPORT
-        assertex(owner==0 && depth==0);
```

Comment:
assert would only be triggered in debug mode (that may or may not be a good thing).
I have only ever hit this when interrupting a program.  I would probably go for a printf, but I'll implement any.  You can choose.